src/prefect/variables.py

```diff
@@ -1,10 +1,130 @@
-from typing import Optional
+from typing import List, Optional
+
+from typing_extensions import Self
 
 from prefect.client.orchestration import PrefectClient
+from prefect.client.schemas.actions import VariableCreate as VariableRequest
+from prefect.client.schemas.actions import VariableUpdate as VariableUpdateRequest
 from prefect.client.utilities import get_or_create_client
 from prefect.utilities.asyncutils import sync_compatible
 
 
+class Variable(VariableRequest):
+    """
+    Variables are named, mutable string values, much like environment variables. Variables are scoped to a Prefect server instance or a single workspace in Prefect Cloud.
+    https://docs.prefect.io/latest/concepts/variables/
+
+    Arguments:
+        name: A string identifying the variable.
+        value: A string that is the value of the variable.
+        tags: A list of strings to associate with the variable.
+    """
+
+    @classmethod
+    @sync_compatible
+    async def set(
+        cls,
+        name: str,
+        value: Optional[str] = None,
+        default: Optional[str] = None,
+        tags: Optional[List[str]] = [],
```

Comment:
It should be okay to pass `None` for tags so this can default to None:

```suggestion
        tags: Optional[List[str]] = None, 
```